sympy/solvers/diophantine.py

```diff
@@ -381,45 +381,113 @@ def diop_linear(eq, param=symbols("t", integer=True)):
 
 
 def _diop_linear(var, coeff, param):
+    """
+    Break down the multivariate equation into multiple
+    bivariate equations of the form:
 
-    x, y = var[:2]
-    a = coeff[x]
-    b = coeff[y]
+    ax + by == d
+
+    which can then be solved using base_solution_linear().
+
+    Example:
+
+    a_0*x_0 + a_1*x_1 + a_2*x_2 == c becomes:
+
+    a_0*x_0 + g_0*y_0 == c 
+
+    where g_0 == gcd(a_1, a_2) and
+          
+          y == a_1*x_1 + a_2*x_2
+               ---       ---
+               g_0       g_0
+
+    Then we can solve for x_0, y_0 with base_solution_linear().
+
+    x_0 is appended to our return value, while y_0 is used to 
+    solve for x_1 and x_2 using base_solution_linear() again.
+    """
+    
+    if len(var) < 2:
+        raise ValueError("Fewer than two elements provided for 'var' at _diop_linear()")
```

Comment:
An equation with a single variable may also have a solution. If it doesn't, ``(None,)`` could be returned. (According to the [documentation](http://docs.sympy.org/latest/modules/solvers/diophantine.html#tutorial), in case of no solutions, a tuple of ``None`` is returned.)